tests/ludwig/utils/test_hyperopt_utils.py

```diff
@@ -0,0 +1,117 @@
+# -*- coding: utf-8 -*-
+# Copyright (c) 2019 Uber Technologies, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+import pytest
+import numpy as np
+from ludwig.utils.hyperopt_utils import GridStrategy, RandomStrategy
+
+
+HYPEROPT_PARAMS = {
+    "test_1": {
+        "params": {
+            "training.learning_rate": {
+                "type": "float",
+                "low": 0.0001,
+                "high": 0.1,
+                "steps": 4,
+                "scale": "log"
+            },
+            "combiner.num_fc_layers": {
+                "type": "int",
+                "low": 0,
+                "high": 4
+            },
+            "utterance.cell_type": {
+                "type": "category",
+                "values": ["rnn", "gru", "lstm"]
+            }
+        },
+        "expected_search_space": {
+            'training.learning_rate': [0.0001, 0.001, 0.01, 0.1],
+            'combiner.num_fc_layers': [0, 4],
```

Comment:
this should probably be [0, 1, 2, 3, 4]

Reply:
Using `np.linspace(start=low,stop=hight,num=steps)` Steps are 2 by default. If we go for using `arange` instead, User won't be sure how much samples could out of a particular param?

Reply:
we can then add a check like:
```
if steps == None:
  steps = high - low + 1
```